src/composition/supergraph/config/resolve/subgraph.rs

```diff
@@ -138,78 +142,153 @@ impl FullyResolvedSubgraph {
                 let supergraph_config_root =
                     supergraph_config_root.ok_or(ResolveSubgraphError::SupergraphConfigMissing)?;
                 let file = unresolved_subgraph.resolve_file_path(supergraph_config_root, file)?;
-                let schema =
-                    Fs::read_file(&file).map_err(|err| ResolveSubgraphError::Fs(Box::new(err)))?;
-                let is_fed_two = schema_contains_link_directive(&schema);
-                Ok(FullyResolvedSubgraph {
-                    routing_url: unresolved_subgraph.routing_url.clone(),
-                    schema,
-                    is_fed_two,
-                })
+                Ok(Self::resolve_file_schema(
+                    unresolved_subgraph.routing_url,
+                    &file,
+                )?)
+            }
+            SchemaSource::SubgraphIntrospection {
+                subgraph_url,
+                introspection_headers,
+            } => Ok(Self::resolve_subgraph_introspection(
+                introspect_subgraph_impl,
+                unresolved_subgraph.name,
+                unresolved_subgraph.routing_url,
+                subgraph_url,
+                introspection_headers,
+            )
+            .await?),
+            SchemaSource::Subgraph {
+                graphref: graph_ref,
+                subgraph,
+            } => Ok(Self::resolve_subgraph(
+                fetch_remote_subgraph_impl,
+                unresolved_subgraph.routing_url,
+                graph_ref,
+                subgraph,
+            )
+            .await?),
+            SchemaSource::Sdl { sdl } => Self::resolve_sdl(sdl),
+        }
+    }
+
+    /// Fully resolves a [`LazilyResolvedSubgraph`] to a [`FullyResolvedSubgraph`]
+    pub async fn fully_resolve(
```

Comment:
our language around resolution is super confusing; `resolve` and `fully_resolve` both return a `FullyResolvedSubgraph`, but the latter makes it sound like "this time I'm serious about fully resolving it" rather than the inputs being different

not sure there's anything to do about it, but it's confusing

Reply:
the way github interleaves the diff makes it hard for me to really understand the differences between the two; probably worth having @dotdat take a stab at it